Source/Details/ASPageTable.h

```diff
@@ -0,0 +1,82 @@
+//
+//  ASPageTable.h
+//  Texture
+//
+//  Copyright (c) 2017-present, Pinterest, Inc.  All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+
+@class ASCollectionElement;
+
+NS_ASSUME_NONNULL_BEGIN
+
+AS_SUBCLASSING_RESTRICTED
+@interface ASPageCoordinates : NSObject
+
+@property (nonatomic, assign, readonly) NSUInteger x;
+@property (nonatomic, assign, readonly) NSUInteger y;
+
++ (instancetype)pageCoordinatesWithX:(NSUInteger)x Y:(NSUInteger)y;
```

Comment:
Good point. At first, `ASPageCoordinate` was actually implemented as a struct. But back then I was not sure if I'd use it frequently enough to justify the NSMapTable and NSArray tricks, so I made it an object. Now that we have a better idea of its usage, let me convert it back to a struct =))